### ISSUE TITLE
Measure how much  memory State takes

### ISSUE THREAD

**Issue by rakita** [bluealloy/revm#611]
This was a feature inside reth where the state was flushed after some memory is taken, this uniforms the size of the bundle that is going to be flushed to db and allows more blocks to be bundles if they are mostly empty (have lower storage write requirement) and with that gain more performance.

There are two things here, One is Cache state size and the second one is Bundle state size. Cache size + bundle size is needed to properly limit the RAM.

**Comment by valo** [comment]
What is a good way to benchmark this? I see that the benches are using `BenchmarkDB`, so that won't work: https://github.com/bluealloy/revm/blob/bd84a077bda907bf4a81133049a2e13f83aceffd/crates/revm/benches/bench.rs#L12. I have one implementation I want to run and measure.

**Comment by valo** [comment]
I am thinking running a reth sync and exposing the mem usage through a promethus metric. Seems like I'll need to wait for https://github.com/paradigmxyz/reth/pull/3512 to be merged first.

**Comment by rakita** [comment]
I meant like adding a memory size that would get incremented when the state gets changed so that we know how much memory the State (Both the bundle and cache) takes, It is not a benchmark per see.

This is rough take in this direction https://github.com/bluealloy/revm/pull/670 but having accurate memory size would be better

**Comment by valo** [comment]
> I meant adding a memory size that would get incremented when the state gets changed so that we know how much memory the State (Both the bundle and cache) takes, It is not a benchmark per see.

Yeah. This is similar to what I implemented, although I am including the size of the structures allocated. What I mean is how to see it in action and may be measure if it properly reflects the process memory changes when the cache is flushed.
